include/veriblock/blockchain/block_status.hpp

```diff
@@ -29,12 +25,23 @@ enum BlockStatus : uint32_t {
       BLOCK_FAILED_CHILD | BLOCK_FAILED_POP | BLOCK_FAILED_BLOCK,
   //! the block has been applied via PopStateMachine
   BLOCK_APPLIED = 1 << 5,
-  //! the block has been at least once successful applied via PopStateMachine
-  BLOCK_CAN_BE_APPLIED = 1 << 6,
-  //! if set, acceptBlock has been executed on this block
-  BLOCK_HAS_PAYLOADS = 1 << 7,
+
+  //! acceptBlockHeader succeded. All ancestors are at least at this state.
+  BLOCK_VALID_TREE = 1 << 6,
+  //! acceptBlock has been executed on this block; payloads are statelessly
+  //! valid
+  BLOCK_HAS_PAYLOADS = 2 << 6,
   //! the block is connected via connectBlock
-  BLOCK_CONNECTED = 1 << 8,
+  BLOCK_CONNECTED = 3 << 6,
+  //! the block has been successfully applied, likely along with another chain
+  BLOCK_HAS_BEEN_APPLIED = 4 << 6,
+  //! the chain with the block at its tip is fully valid
+  BLOCK_CAN_BE_APPLIED = 5 << 6,
```

Comment:
I don't think you can build flags like this. For example BLOCK_VALID_TREE | BLOCK_HAS_BEEN_APPLIED equals BLOCK_CAN_BE_APPLIED.

Reply:
BLOCK_VALID_TREE and BLOCK_CAN_BE_APPLIED aren't flags as such,  they are mutually exclusive. Effectively we have a 3-bit uint here.

Reply:
thinking more about it, maybe we should use a proper bit field here. will discuss it with Bohdan

Reply:
Yes I think they should be separate from flags to avoid confusion.